io/GDALReader.cpp

```diff
@@ -112,15 +121,41 @@ void GDALReader::addDimensions(PointLayoutPtr layout)
 {
     layout->registerDim(pdal::Dimension::Id::X);
     layout->registerDim(pdal::Dimension::Id::Y);
-    for (int i = 0; i < m_raster->bandCount(); ++i)
+
+    if (m_header.size())
```

Comment:
The cerr stuff needs to go.

Also, IMO would be cleaner if you just assigned band-X... to dimNames if it was empty.  Then you only need one loop to assign dims.

Reply:
Done in d0f7d0646c04baf68249a2ba60dc51971f21c97b